web/src/components/CVPreview/CVPreview.jsx

```diff
@@ -0,0 +1,429 @@
+                [field]: value
+            }
+        }));
+    };
+
+    const handleArrayChange = (arrayName, index, field, value) => {
+        setEditedData(prev => ({
+            ...prev,
+            [arrayName]: prev[arrayName].map((item, i) => 
+                i === index ? { ...item, [field]: value } : item
+            )
+        }));
+    };
+
+    const handleBulletPointChange = (expIndex, bulletIndex, value) => {
+        setEditedData(prev => ({
+            ...prev,
+            experience: prev.experience.map((exp, i) => 
+                i === expIndex ? {
+                    ...exp,
+                    bulletPoints: exp.bulletPoints.map((point, j) => 
+                        j === bulletIndex ? value : point
+                    )
+                } : exp
+            )
+        }));
+    };
+
+    const addExperience = () => {
+        setEditedData(prev => ({
+            ...prev,
+            experience: [...(prev.experience || []), {
+                position: '',
+                company: '',
+                date: '',
+                bulletPoints: ['']
+            }]
+        }));
+    };
+
+    const addProject = () => {
+        setEditedData(prev => ({
+            ...prev,
+            projects: [...(prev.projects || []), {
+                name: '',
+                description: ''
+            }]
+        }));
+    };
+
+    const addEducation = () => {
+        setEditedData(prev => ({
+            ...prev,
+            education: [...(prev.education || []), {
+                degree: '',
+                institution: '',
+                year: ''
+            }]
+        }));
+    };
+
+    const removeItem = (arrayName, index) => {
+        setEditedData(prev => ({
+            ...prev,
+            [arrayName]: prev[arrayName].filter((_, i) => i !== index)
+        }));
+    };
+
+    const addBulletPoint = (expIndex) => {
+        setEditedData(prev => ({
+            ...prev,
+            experience: prev.experience.map((exp, i) => 
+                i === expIndex ? {
+                    ...exp,
+                    bulletPoints: [...(exp.bulletPoints || []), '']
+                } : exp
+            )
+        }));
+    };
+
+    const removeBulletPoint = (expIndex, bulletIndex) => {
+        setEditedData(prev => ({
+            ...prev,
+            experience: prev.experience.map((exp, i) => 
+                i === expIndex ? {
+                    ...exp,
+                    bulletPoints: exp.bulletPoints.filter((_, j) => j !== bulletIndex)
+                } : exp
+            )
+        }));
+    };
+
+    const handleSave = () => {
+        setIsEditing(false);
+        if (onSave) {
+            onSave(editedData);
+        }
+    };
+
+    const handleCancel = () => {
+        setEditedData(cvData);
+        setIsEditing(false);
+    };
+
+    const { fullName, contact, professionalSummary, experience, projects, education, yourProfile_vs_jobCriteria } = isEditing ? editedData : cvData;
```

Comment:
the `professionalSummary` variable here does not match the `professional_summary` in the preview page and will cause this variable to not be displayed